### PR TITLE
Check React keys in the production build

### PR DESCRIPTION
**Pull request by lukebrody**
React validates keys only in its development build, so a duplicate or missing key ships to the live site with nothing to report it. This aliases `react` to a shim that inspects each array child as it is created, which puts the same check in both builds. The messages carry `[failtest]`, so the e2e harness fails on them — including missing keys, which have never failed a test here: `test_utils` matched React's own duplicate-key message, and that one only exists in development. That match is now gone, since the shim covers both builds.

The shim reports one problem per list, so a long list can't flood the console on every render, and it stays quiet for the cases React scopes by position: a nested array and a separate positional child each get a key of their own.

Two things to expect. The first CI run may turn up pre-existing missing keys that nothing has been catching. And the check costs a `Set` per array child per render; if that ever shows up in a profile, gate it on a flag read once at module load.

`react-duplicate-keys.cjs` is CommonJS because it re-exports everything React exports, which ESM can't do statically over a CJS module, and it reaches React by path because the alias would otherwise point back at itself. `.cjs` joins `.js` and `.mjs` in the eslint ignore list, since a file whose types come from an untyped CJS module can't satisfy the type-aware rules.

The e2e test triggers the failure through JSX, in `bad-keys.tsx` — it is what real code compiles to, and it is the only way to confirm the alias is live in the served bundle rather than just testing the checking function.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: react/rspack.config.js [hunk 1]
```diff
@@ -32,6 +32,9 @@ export default env => ({
         clean: true,
     },
     resolve: {
+        alias: {
+            react$: path.resolve(import.meta.dirname, 'src/react-duplicate-keys.cjs'),
```
[the remark this answers is not in the document]
Comment (lukebrody) [reply react/rspack.config.js:36]: The build analysis is right — the alias is unconditional and `build.py` runs `npm run prod` for both `ci` and `prod`, so the live site gets the shim. That part is intentional: React validates keys only in its development build, so without this a bad key ships to urbanstats.org unnoticed. The `[failtest]` marker reaching real consoles is acceptable; a key collision on the live site is something I want to hear about.

The perf cost is too small to trade that coverage away. Measured on node 24 against React 18's production build:

| | ns per `createElement` |
|---|---|
| plain | 22 |
| shim | 66 |

3x on `createElement` itself, or +44ns per element. In a fuller test — `renderToString` of a 300x12 table, ~11k elements — the shim added 0.9ms to a 2.0ms render. In a browser that render is dominated by reconciliation, layout, and paint, so a 10k-element page pays something like half a millisecond.

One correction to the cost model: the `Set` per mapped list is the cheap part, ~17ns per element. Most of the 44ns is `positional.add(child)`, a `WeakSet` insert for every positional child at ~39ns. Not a reason to change anything, just where the time goes.

No changes here.
